src/PostRector/Rector/UnusedImportRemovingPostRector.php

```diff
@@ -40,22 +40,34 @@ public function enterNode(Node $node): ?Node
             ? $node->name
             : null;
 
-        $names = $this->resolveUsedPhpAndDocNames($node);
+        $namesInOriginalCase = $this->resolveUsedPhpAndDocNames($node);
+        $namesInLowerCase = null; // Initialized as null, lazy loaded when case in-sensitive names are needed
 
         foreach ($node->stmts as $key => $namespaceStmt) {
             if (! $namespaceStmt instanceof Use_) {
                 continue;
             }
 
-            if ($namespaceStmt->uses === []) {
+            if ($namespaceStmt->uses === [] || $namesInOriginalCase === []) {
                 unset($node->stmts[$key]);
                 $hasChanged = true;
 
                 continue;
             }
 
             $useUse = $namespaceStmt->uses[0];
```

Comment:
before check on `0` index, probably better to check `count()` uses first, to avoid invalid removal on : 

```php
namespace A;

use B, C;

echo B::class;
```

see https://getrector.com/demo/755f134f-8f83-4bb1-a904-99ab62db0c6b

with check:

```suggestion
+           if (count($namespaceStmt->uses) > 1) {
+                   continue;
+           }
            
            $useUse = $namespaceStmt->uses[0];
```

Reply:
Nice finding. I can handle it in another PR. It will be easier to review the changes. What do you think?

Reply:
sure 👍